addons/addon-base-raas/packages/base-raas-services/lib/study/study-service.js

```diff
@@ -526,6 +527,21 @@ class StudyService extends Service {
     return result;
   }
 
+  /**
+   * IMPORTANT: Do NOT call this method directly from a controller, this is because
+   * this method does not do any authorization check.  It will return the study given
+   * a study id no matter who the requestContext principal is.
+   */
+  async listBucketsForAccount(requestContext, { accountId }) {
```

Comment:
Are you trying to list buckets that belong to a data source account?
If so, the dataSourceAccountService already has this method.  The study service does not know how to get a list of buckets given a data source account.

Reply:
Sorry, the name for this method is listStudiesForAccount. This will get all studies that are linked to a given accountId.